index.bs

```diff
@@ -2712,7 +2591,181 @@ partial interface MLGraphBuilder {
+        1. Set |shapeOutput|[|rankOutput| + |dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Set |rankOutput| to |rankOutput| + |dimCount|.
+    1. Let |dimCount| be zero.
+    1. [=map/For each=] |size| &rarr; |value| of |shapeInput|:
+        1. If |dimCount| is less than or equal to |axis| then [=continue=].
+        1. Set |shapeOutput|[|rankOutput| + |dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Let |desc| a new {{MLOperandDescriptor}}.
+    1. Set |desc|.{{MLOperandDescriptor/dimensions}} to |shapeOutput|.
+    1. Set |desc|.{{MLOperandDescriptor/type}} to |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/type}}.
+    1. If any of the following sub-steps fail, [=exception/throw=] an "{{OperationError}}" {{DOMException}}.
+        1. Let |output| be the result of <a>creating an MLOperand</a> given |desc|.
+        1. Make a request to the underlying platform to:
+            1. Let |opImpl| be an [=implementation-defined=] platform operator for the Gather operation, given |input|, |indices|, and |options|.
+            1. Store a reference of |opImpl| in |output|.{{MLOperand/[[operator]]}}.
+            1. Create an [=implementation-defined=] platform operand |outputImpl| to represent the output, given |output| and |opImpl|.
+            1. Store a reference to |outputImpl| in |output|.{{MLOperand/[[operand]]}}.
+        1. Connect |input|.{{MLOperand/[[operand]]}} and |indices|.{{MLOperand/[[operand]]}} as inputs to |opImpl|.
+        1. Connect |output|.{{MLOperand/[[operand]]}} as output to |opImpl|.
+    1. Return |output|.
+  </div>
+</details>
+
+<div class="example">
+<details open>
+  <summary>
+    Examples of how gather works in different slicing schemes.
+  </summary>
+  <pre highlight="js">
+    // input of shape [4,3]:
+    //   [[ 0,  1,  2],
+    //    [10, 11, 12], 
+    //    [20, 21, 22], 
+    //    [30, 31, 32]]
+    const input = builder.constant(
+    { dimensions: [4,3] }, new Float32Array([0,1,2,10,11,12,20,21,22,30,31,32]));
+
+    const indices1 = builder.constant(
+    { type: 'uint32', dimensions: [2] }, new Uint32Array([3,1]));
+
+    const indices2 = builder.constant(
+    { type: 'uint32', dimensions: [3] }, new Uint32Array([2,1,1]));
+
+    const indices3 = builder.constant(
+    { type: 'uint32', dimensions: [2,2] }, new Uint32Array([0,1,1,2]));
+
+    // axis = 0 (default)
+    // indices of shape [2]: 
+    //   [3,1]
+    // output of shape [2,3]:
+    //   [[30, 31, 32], 
+    //    [10, 11, 12]]
+    const output1 = builder.gather(input, indices1);
+
+    // axis = 1
+    // indices of shape [3]:
+    //   [2,1,1]
+    // output of shape [4,3]:
+    //   [[ 2,  1,  1],
+    //    [12, 11, 11], 
+    //    [22, 21, 21],
+    //    [32, 31, 31]]
+    const output2 = builder.gather(input, indices2, { axis: 1 });
+
+    // axis = 1
+    // indices of shape [2,2]: 
+    //   [[0, 1], 
+    //    [1, 2]]
```

Comment:
Should we add a bit more extreme example?

```
input.shape = (2,3,2)
indices.shape = (3,4,5)
output.shape = (2,3,4,5,2)
```

I think this would work per definition?

Reply:
@wacky6 thanks for the proposal. Exemplary examples in the specs are important. Please open a separate issue so the WG can review and make possible improvements also to other examples at the same time.

I consider this non-blocking for this PR; I will thus mark this conversation resolved.